src/jupyterlabcontroller/services/fileserver.py

```diff
@@ -46,142 +67,261 @@ def __init__(
         logger: BoundLogger,
     ) -> None:
         self._config = config
         self._builder = fileserver_builder
         self._storage = fileserver_storage
+        self._slack = slack_client
         self._logger = logger
 
         if not self._config.enabled:
             raise NotConfiguredError("Fileserver is disabled in configuration")
 
-        self._user_map = FileserverUserMap()
-        self._tasks: set[asyncio.Task] = set()
-        self._started = False
-
-        # This maps usernames to locks, so we have a lock per user, and if
-        # there is no lock for that user, requesting one gets you a new lock.
-        self._lock: dict[str, asyncio.Lock] = defaultdict(asyncio.Lock)
+        # Background task management.
+        self._scheduler: Scheduler | None = None
 
-        # Maps users to the tasks watching for their pods to exit
-        self._watches: set[asyncio.Task] = set()
+        # Mapping of usernames to internal state.
+        self._servers: dict[str, _State] = {}
 
     async def create(self, user: UserInfo) -> None:
-        """If the user doesn't have a fileserver, create it.  If the user
-        already has a fileserver, just return.
+        """Ensure a file server exists for the given user.
+
+        If the user doesn't have a fileserver, create it.  If the user already
+        has a fileserver, just return. This gets called by the handler when a
+        user comes in through the ``/files`` ingress.
 
-        This gets called by the handler when a user comes in through the
-        /files ingress.
+        Parameters
+        ----------
+        user
+            User for which to create a file server.
+
+        Raises
+        ------
+        KubernetesError
+            Raised if there is some failure in a Kubernetes API call.
         """
-        username = user.username
-        self._logger.info(f"Fileserver requested for {username}")
-        if not await self._user_map.get(username):
+        logger = self._logger.bind(user=user.username)
+        self._logger.info("File server requested")
+        if user.username not in self._servers:
+            self._servers[user.username] = _State(running=False)
+        state = self._servers[user.username]
+        async with state.lock:
+            if state.running:
+                return
             try:
-                await self._create_fileserver(user)
-            except Exception:
-                msg = (
-                    f"Fileserver creation for {username} failed, deleting"
-                    " fileserver objects"
-                )
-                self._logger.exception(msg)
-                await self.delete(username)
+                await self._create_file_server(user)
+            except Exception as e:
+                logger.exception("File server creation failed")
+                await self._maybe_post_slack_exception(e, user.username)
+                logger.info("Cleaning up left-over file server objects")
```

Comment:
Because I can't decide whether it's "left over", "left-over", or "leftover", maybe replace with "orphaned" ?